src/lib/ts-helpers.ts

```diff
@@ -575,3 +575,37 @@ export function createEnum(name: string, values: (string | number)[]) {
         members
     );
 }
+
+export function createUnionTypeValues(
+    name: string,
+    values: (string | number)[]
+) {
+    const props = values.map(v =>
+        ts.factory.createPropertyAssignment(
+            v.toString(),
+            typeof v === "string"
+                ? ts.factory.createStringLiteral(v)
+                : ts.factory.createNumericLiteral(v)
+        )
+    );

```

Comment:
[IMO] How about prefixing numeric keys with an `_` to allow property access with dot notation?

```ts
    const props = values.map(v =>
        ts.factory.createPropertyAssignment(
            typeof v === "number" ? "_" + v.toString() : v.toString(),
            typeof v === "string"
                ? ts.factory.createStringLiteral(v)
                : ts.factory.createNumericLiteral(v)
        )
    );
```

Expected output would be:
```ts
    export const NumberEnum = {
       _1: 1,
       _2: 2
    } as const;
```

Now members can be access normally like `NumberEnum._1` instead of `NumberEnum["1"]`

Reply:
Looks good, even if you define a number as the key of the Object, it will actually be treated as a string. So, it is a good idea to add an underscore at the beginning to make it clear that it is a string.
I'll try to fix it.

Reply:
9c16774
Fixed.